packages/core/client/src/route-switch/RouteSwitch.tsx

```diff
@@ -1,59 +1,90 @@
-import React from 'react';
-import { Redirect, Route, Switch } from 'react-router-dom';
+import React, { useCallback } from 'react';
+import { Navigate, useRoutes } from 'react-router-dom';
 import { RouteContext } from './context';
 import { useRouteComponent } from './hooks';
 import { RouteSwitchProps } from './types';
 
 export function RouteSwitch(props: RouteSwitchProps) {
-  const { routes = [] } = props;
-  if (!routes.length) {
-    return null;
-  }
-  return (
-    <Switch>
-      {routes.map((route, index) => {
-        if (route.type == 'redirect') {
-          return (
-            <Redirect
-              key={index}
-              to={route.to}
-              push={route.push}
-              from={route.from}
-              path={route.path}
-              exact={route.exact}
-              strict={route.strict}
-            />
-          );
-        }
-        if (!route.path && Array.isArray(route.routes)) {
-          route.path = route.routes.map((r) => r.path) as any;
+  const { routes: remoteRoutes = [] } = props;
+
+  /**
+   * 将 path 中的可选参数转换为多个路径
+   *
+   * @example
+   * generatePaths('/a/b/:c?/:d?')
+   * => ['/a/b', '/a/b/:c', '/a/b/:c/:d']
+   */
+  const generatePaths = useCallback((path = '') => {
+    if (path === '/') return [path];
+
+    const segments: string[] = path.split('/'); // 将路径按照 '/' 分段
+    if (segments[segments.length - 1] === '') {
+      segments.pop(); // 最后一个字符是 '/'，将其弹出
+    }
+    let lastOptionsIndex = undefined;
+    for (let i = segments.length - 1; i >= 0; i--) {
+      if (!segments[i].endsWith('?')) {
+        lastOptionsIndex = i; // 找到最后一个可选参数的位置
+        break;
+      }
+    }
+
+    if (lastOptionsIndex === undefined) return [path]; // 没有可选参数，直接返回原路径
+    const res: string[] = [];
+    for (let i = lastOptionsIndex; i < segments.length; i++) {
+      res.push(
+        segments
+          .slice(0, i + 1)
+          .join('/')
+          .replace(/\?/g, '')
+          .replace(/\(\.\+\)/g, ''), // 历史遗留问题 /a/b/:c(.+)? => /a/b/:c
+      );
+    }
+    return res;
+  }, []);
+
+  const getRoutes = useCallback(
+    (routes: RouteSwitchProps['routes']) => {
+      const res = routes.map((item) => {
+        if (item.type === 'route' && item.routes) {
+          // layout
+          return {
+            element: (
+              <RouteContext.Provider value={item}>
+                <ComponentRenderer component={item.component} />
+              </RouteContext.Provider>
+            ),
+            children: getRoutes(item.routes).flat(Infinity),
+          };
+        } else if (item.type === 'route' && item.path) {
```

Comment:
```suggestion
        }
        if (item.type === 'route' && item.path) {
```
感觉把 `else` 去掉会更简洁一点。

Reply:
`else if` 主要是强调非此即彼的认知，如果只是多个 `if`，达不到这个感觉，当然每个人的风格略有不同，我相对比较喜欢这种。